app/lib/features/onboarding/widgets/invite_friends_widget.dart

```diff
@@ -47,19 +48,86 @@ class _InviteFriendsWidgetState extends ConsumerState<InviteFriendsWidget> {
                     const SizedBox(height: 10),
                     _buildSearchBar(context),
                     const SizedBox(height: 10),
+                    if (searchValue == null || searchValue.isEmpty)
+                      _buildInviteExternallyWidget(context),
                     _buildSuggestedUsers(context, widget.roomId),
                   ],
                 ),
               ),
               _buildActionButton(context, L10n.of(context)),
-              const SizedBox(height: 50),
+              const SizedBox(height: 20),
             ],
           ),
         ),
       ),
     );
   }
 
+  Widget _buildInviteExternallyWidget(BuildContext context) {
+    final lang = L10n.of(context);
+    final textTheme = Theme.of(context).textTheme;
+    final inviteCodeList = ref.watch(superInvitesForRoom(widget.roomId)).valueOrNull ?? [];
+    String inviteCode;
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 10),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(
+            lang.inviteExternally,
+            textAlign: TextAlign.center,
+            style: textTheme.bodyLarge,
+          ),
+          const SizedBox(height: 10),
+          Row(
+            children: [
+              _buildInviteExternallyIcon(
+                context, 
+                PhosphorIcons.copy(),
+                () async {
+                  inviteCode = await getInviteCode(context, inviteCodeList, ref, widget.roomId);
```

Comment:
```suggestion
                  inviteCode = await getInviteCode(context, ref, widget.roomId);
```

`inviteCodeList` should be fetch in function itself, why to pass it here?